test/tools/Modules/HttpsListener/HttpsListener.psm1

```diff
@@ -0,0 +1,126 @@
+Class HttpsListener 
+{
+    [int]$Port
+    [System.Management.Automation.Job]$Job
+    
+    HttpsListener () { }
+
+    [string] ToString() {
+        return ('Port: {0}; Status: {1}' -f $This.Port, $This.JobStateInfo.State)
+    }
+}
+
+$HttpsListeners = @{}
+
+function Get-HttpsListener 
+{
+    [CmdletBinding(ConfirmImpact = 'Low')]
+    [OutputType([HttpsListener])]
+    param 
+    (
+        [ValidateRange(1,65535)]
+        [int[]]$Port
+    )
+
+    process 
+    {
+        if ($Port)
+        {
+            return $Script:HttpsListeners[$port]
```

Comment:
@iSazonov  Assuming we may need multiple HTTPS based listeners with different certificates. For example, we may want a listener with an expired certificate to test expired cert scenarios vs self-signed vs invalid chain, etc. For an HTTP (no S) listener, 1 port and 1 listener works, but for HTTPS it is probably not.